### PR TITLE
gen_cli: Replace underscores by dashes for the generated cmd arguments

### PR DESCRIPTION
**Pull request by hectorj2f**
Related to this issues: https://github.com/goadesign/goa/issues/1035

@raphael Please, have a look! I made some initial changes that works in my branch. But I believe I might miss stuff. Also, I was thinking to pass a different parameter into the map instead of modifying the `Parent.Name` property.

ping @raphael

### CODE REVIEW

File: goagen/codegen/helpers.go [hunk 1]
```diff
@@ -164,3 +164,38 @@ func SnakeCase(name string) string {
 	}
 	return b.String()
 }
+
+// KebabCase produces the kebab-case version of the given CamelCase string.
+func KebabCase(name string) string {
```
Comment (raphael) [inline goagen/codegen/helpers.go]: Couldn't this simply call `SnakeCase` and then replace the `_` with `-`?

File: goagen/gen_client/cli_generator.go [hunk 2]
```diff
@@ -162,6 +162,9 @@ func (g *Generator) generateCommands(commandsFile string, clientPkg string, func
 		}
 		return res.IterateActions(func(action *design.ActionDefinition) error {
 			name := codegen.Goify(action.Name, false)
+			// To avoid underscore in the resource name, we apply the kebabCase
+			// algorithm over the action.Parent.Name property.
+			action.Parent.Name = codegen.KebabCase(action.Parent.Name)
```
Comment (raphael) [inline goagen/gen_client/cli_generator.go]: I think a better approach here would be to leave the name of the resource and action unchanged (they could be used for other things now and in the future) and instead call the `KebabCase` function from the template where the command and action names are defined (see `registerCmdsT`).

File: goagen/codegen/helpers.go (hunk 1 above)
Comment (hectorj2f) [inline goagen/codegen/helpers.go]: Sure.

File: goagen/gen_client/cli_generator.go (hunk 2 above)
Comment (hectorj2f) [inline goagen/gen_client/cli_generator.go]: Alright!

**Comment by hectorj2f** [comment]
@raphael 👍, I'll add a test for the KebabCase function

File: goagen/codegen/helpers_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	"testing"
+
+	"github.com/goadesign/goa/goagen/codegen"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestHelpers(t *testing.T) {
+	var (
+		stringTestA = "testAa"
```
Comment (kwk) [inline goagen/codegen/helpers_test.go]: Why not inline those to the `Expect` call below? It would simplify reading the tests a bit.

Comment (hectorj2f) [inline goagen/codegen/helpers_test.go]: I wanted to be specific but I have nothing against this suggestion :). I'll apply those changes

Comment (hectorj2f) [inline goagen/codegen/helpers_test.go]: Done @kwk

**Comment by hectorj2f** [comment]
@raphael I addressed your comment, tests seem to be happy :).

**Comment by raphael** [comment]
Looks great, thank you!

**Comment by raphael** [comment]
Do you think you could backport these changes to the `v1` branch?

**Comment by hectorj2f** [comment]
`Do you think you could backport these changes to the v1 branch?` I won't know how to do it. Could you ?

**Comment by raphael** [comment]
OK :) FYI all that is required is to checkout the `v1` branch, cherry-pick the commit and create a new PR against the `v1` branch. I'll do it.

**Pull request merged**